man/sd_notify.xml

```diff
@@ -205,25 +205,23 @@
-        This is useful for implementing service restart schemes where services serialize
-        their state to <filename>/run</filename>, push their file descriptors to the
-        system manager, and are then restarted, retrieving their state again via socket
-        passing and <filename>/run</filename>. Note that the service manager will accept
-        messages for a service only if <varname>FileDescriptorStoreMax=</varname> is set
-        to non-zero for it (defaults to zero, see
-        <citerefentry><refentrytitle>systemd.service</refentrytitle><manvolnum>5</manvolnum></citerefentry>).
-        File descriptors must be pollable, see
-        <citerefentry><refentrytitle>epoll_ctl</refentrytitle><manvolnum>2</manvolnum></citerefentry>.
-        Multiple arrays of file descriptors may be sent in separate messages, in which
-        case the arrays are combined.  Note that the service manager removes duplicate
-        file descriptors before passing them to the service. Use
-        <function>sd_pid_notify_with_fds()</function> to send messages with
-        <literal>FDSTORE=1</literal>, see below.</para></listitem>
+        <listitem><para>Stores additional file descriptors in the service manager. File descriptors sent this way will
+        be maintained per-service by the service manager and will be passed again using the usual file descriptor
+        passing logic on the next invocation of the service, see
+        <citerefentry><refentrytitle>sd_listen_fds</refentrytitle><manvolnum>3</manvolnum></citerefentry>.  This is
+        useful for implementing service restart schemes where services serialize their state to a
+        <citerefentry><refentrytitle>memfd_create</refentrytitle><manvolnum>2</manvolnum></citerefentry> memory file
+        descriptor, push this and their other file descriptors to the system manager, and are then restarted,
+        retrieving their state again via socket passing. Note that the service manager will accept messages for a
+        service only if <varname>FileDescriptorStoreMax=</varname> is set to non-zero for it (defaults to zero, see
+        <citerefentry><refentrytitle>systemd.service</refentrytitle><manvolnum>5</manvolnum></citerefentry>). If file
+        descriptors sent are pollable (see
+        <citerefentry><refentrytitle>epoll_ctl</refentrytitle><manvolnum>2</manvolnum></citerefentry>), then any
+        <constant>EPOLLHUP</constant> or <constant>EPOLLERR</constant> event seen on them will result in their removal
+        from the store. Multiple arrays of file descriptors may be sent in separate messages, in which case the arrays
+        are combined. Note that the service manager removes duplicate file descriptors before passing them to the
+        service. Use <function>sd_pid_notify_with_fds()</function> to send messages with <literal>FDSTORE=1</literal>,
+        see below.</para></listitem>
```

Comment:
I have just read https://github.com/systemd/systemd/issues/5606. The following sentence
> The API returns success but later on in the logs I see that systemd is complaining about the received file descriptor.

makes me think that some users might assume that `sd_pid_notify_with_fds` does not return success when `fd` has not been stored. This is incorrect assumption because `sd_pid_notify_with_fds` simply checks that `sendmsg` was successful. Probably, this should be documented more clearly than

> If the status was sent, these functions return with a positive return value.